Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -13,12 +13,35 @@ public protocol Rule {
     func validateFile(file: File) -> [StyleViolation]
 }
 
+extension Rule {
```

Comment:
Well it would be possible. The problem is that making `Rule` conform to `Equatable` will impose a `Self` requirement on `Rule`, therefore you won't be able to use it as a type (e.g. `[Rule]`) and only as a generic constraint. This exact case is covered really well in the 2015 WWDC session on Protocol Orientated Programming, and implementing `isEqualTo` for situations like this was the recommended solution. Now, that still doesn't stop us from implementing `==` as well using `isEqualTo`, and I originally did that, but then took it out as because these sort of semantics should probably be discussed before implemented just because it's not 100% clear. In addition to the WWDC video, there's a pretty decent discussion of it [here](http://www.llamagraphics.com/content/combining-swift-protocols). I can definitely implement it with `Equatable`, but I'll have to modify all the cases that the code uses `Rule` as a type.